packages/amplication-server/src/core/entity/entity.service.spec.ts

```diff
@@ -646,6 +664,54 @@ describe("EntityService", () => {
     expect(prismaEntityFieldCreateMock).toBeCalledTimes(3);
   });
 
+  describe("service license", () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    describe("when billing is not enabled", () => {
+      beforeEach(() => {
+        billingServiceIsBillingEnabledMock.mockReturnValue(false);
+      });
+
+      it("should not throw an error even when the project or the service is not licensed", async () => {
+        const resource = {
+          ...EXAMPLE_RESOURCE,
+          licensed: false,
+          project: { ...EXAMPLE_RESOURCE.project, licensed: false },
+        };
+        await expect(
+          service.checkServiceLicense(resource)
+        ).resolves.not.toThrow(BillingLimitationError);
+      });
+    });
+
+    describe("when billing is enabled", () => {
+      beforeEach(() => {
+        billingServiceIsBillingEnabledMock.mockReturnValue(true);
+      });
+
+      it("should not throw an error when project or service is under license", async () => {
+        await expect(
+          service.checkServiceLicense(EXAMPLE_RESOURCE) // in the example resource the project and the service are licensed
+        ).resolves.not.toThrow(BillingLimitationError);
+      });
+
+      it("should throw an error when project or service is not under license", async () => {
+        const resource = {
+          ...EXAMPLE_RESOURCE,
+          licensed: false,
```

Comment:
We can add a separate test for each of the licenses (one with project.licensed=F and resource.licensed=T and one test with the opposite). Not mandatory, for your consideration. I will approve the PR anyway and you can add this if you want.

Reply:
if project.licensed=false and resource.licensed=true => we probably did something wrong, but I can add this test because it's a valid scenario based on the function.